File: 001_Python/20211011_1840_Hackerrank_Pangrams_Easy/pangrams_v01.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'pangrams' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def pangrams(s):
    # Write your code here
    letters = set()
    #print(f'len(s) = {len(s)}')
    if len(s) < 26:
        return "not pangram"
    for c in s.lower():
        if c != " ":
            letters.add(c)
    #print(f'len(letters) = {len(letters)}')
    if len(letters) == 26:
        return "pangram"
    return "not pangram"

    
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = pangrams(s)

    fptr.write(result + '\n')

    fptr.close()

File: 003_C++/20211011_1840_Hackerrank_Pangrams_Easy/pangrams_v01.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'pangrams' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string pangrams(string s) {
    vector<bool> letters(26,false);
    int sum(0);
    int index;
    int offset = (int)'a';
    for(size_t i=0; i<s.size(); i++){
        if(s[i] != ' '){
            index = (int)tolower(s.at(i)) - offset;
            cout << "index" << index << endl;
            if(letters[index] == false){
                letters[index] = true;
                sum++;
            }
        }
    }
    
    if(sum != 26){
        return "not pangram";
    }
    
    return "pangram";
    
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = pangrams(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
